openmetadata-ui/src/main/resources/ui/cypress/e2e/AddNewService/redshiftWithDBT.spec.js

```diff
@@ -0,0 +1,169 @@
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import {
+    deleteCreatedService,
+    editOwnerforCreatedService,
+    goToAddNewServicePage,
+    interceptURL,
+    login, testServiceCreationAndIngestion,
+    updateDescriptionForIngestedTables,
+    uuid,
+    verifyResponseStatusCode,
+    visitEntityDetailsPage
+} from '../../common/common';
+import {
+    DBT, HTTP_CONFIG_SOURCE, LOGIN,
+    SERVICE_TYPE
+} from '../../constants/constants';
+
+const serviceType = 'Redshift';
+const serviceName = `${serviceType}-ct-test-${uuid()}`;
+const tableName = 'boolean_test';
+const DBTTable = 'customers';
+const description = `This is ${serviceName} description`;
+
+describe('RedShift Ingestion', () => {
+  beforeEach(() => {
+    login(LOGIN.username, LOGIN.password);
+    cy.goToHomePage();
+  });
+  it('add and ingest data', () => {
+    goToAddNewServicePage(SERVICE_TYPE.Database);
+    const connectionInput = () => {
+      cy.get('#root_username').type(Cypress.env('redshiftUsername'));
+      cy.get('#root_password')
+        .scrollIntoView()
+        .type(Cypress.env('redshiftPassword'));
+      cy.get('#root_hostPort')
+        .scrollIntoView()
+        .type(Cypress.env('redshiftHost'));
+      cy.get('#root_database')
+        .scrollIntoView()
+        .type(Cypress.env('redshiftDatabase'));
+    };
+
+    const addIngestionInput = () => {
+      // no schema or database filters
+      cy.get('[data-testid="schema-filter-pattern-checkbox"]').check();
+      cy.get('[data-testid="filter-pattern-includes-schema"]')
+        .should('be.visible')
+        .type('dbt_jaffle');
+      cy.get('[data-testid="toggle-button-include-views"]')
+        .should('be.visible')
+        .click();
+    };
+
+    const configureDBT = () => {
+      cy.contains('Configure DBT Model').should('be.visible');
+      cy.get('[data-testid="dbt-source"]')
+        .should('be.visible')
+        .select('HTTP Config Source');
+      cy.get('[data-testid="catalog-url"]')
+        .scrollIntoView()
+        .should('be.visible')
+        .type(HTTP_CONFIG_SOURCE.DBT_CATALOG_HTTP_PATH);
+      cy.get('[data-testid="manifest-url"]')
+        .scrollIntoView()
+        .should('be.visible')
+        .type(HTTP_CONFIG_SOURCE.DBT_MANIFEST_HTTP_PATH);
+      cy.get('[data-testid="run-result-file"]')
+        .scrollIntoView()
+        .should('be.visible')
+        .type(HTTP_CONFIG_SOURCE.DBT_RUN_RESTLTS_FILE_PATH);
+    };
+
+    testServiceCreationAndIngestion(
+      serviceType,
+      connectionInput,
+      addIngestionInput,
+      serviceName,
+      'database',
+      true,
+      configureDBT
```

Comment:
Not sure why we are passing it we can call it manually post this method runs successfully.
Let me know if we have any solid reason to put it as callback here

Reply:
So DBT config step can differ from connector to connector and its data as well, so to perform dynamic logic into the common flow we are passing the function!

Reply:
What's challenge in calling it after this file it self?

Reply:
Since it's intermediate step we need it as callback function